geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/geotools/GeoMesaFeatureWriter.scala

```diff
@@ -51,8 +49,13 @@ trait GeoMesaFeatureWriter[DS <: GeoMesaDataStore[DS]] extends SimpleFeatureWrit
     // `write` will calculate all mutations up front in case the feature is not valid, so we don't write partial entries
     try { getWriter(writable).write(writable, update) } catch {
       case NonFatal(e) =>
-        val attributes = s"${writable.getID}:${writable.getAttributes.asScala.mkString("|")}"
-        throw new IllegalArgumentException(s"Error indexing feature '$attributes'", e)
+        val msg = s"Error indexing feature '${writable.getID}:${DataUtilities.encodeFeature(writable, false)}'"
+        if (e.isInstanceOf[IllegalArgumentException]) {
+          throw new IllegalArgumentException(msg, e)
+        } else {
+          throw new RuntimeException(msg, e)
```

Comment:
Ooo, this is new.  What gives us a non-IllegalArgumentException?

Reply:
errors in e.g. writing a mutation to hbase (table offline, timeout, whatever). i think this code originally just generated the mutations, but didn't write them, so throwing IllegalArgumentException was correct. it was updated at some point to both generate mutations and write them, but the catch/throw wasn't updated. admittedly this solution isn't great, but it requires the least change. someone on gitter reported an error thrown by this block which made me check it again.

Reply:
Perfect!